ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -617,49 +617,58 @@ public class Manager: NSObject, UIGestureRecognizerDelegate {
       throw KMPError.wrongPackageType
     }
 
-    for k in kmp.keyboards {
-      let installableKeyboards : [InstallableKeyboard] = k.installableKeyboards
-      let keyboardID = k.keyboardId
+    for r in kmp.resources {
+      let installableFiles: [(LanguageResource, [(String, URL)])] = r.installableResources.map { resource in
+        // (source file, destination file)
+        var set: [(String, URL)] = [(resource.sourceFilename, Storage.active.resourceURL(for: resource)!)]
+        let installableFonts: [(String, URL)] = resource.fonts.map { font in
+          // KMPs only list a single font file for each entry whenever one is included.
+          // A pre-existing assumption.
+          let fontFile = font.source[0]
+          return (fontFile, Storage.active.fontURL(forResource: resource, filename: fontFile)!)
+        }
 
-      do {
-        try FileManager.default.createDirectory(at: Storage.active.keyboardDir(forID: keyboardID),
-                                                withIntermediateDirectories: true)
-      } catch {
-        log.error("Could not create dir for download: \(error)")
-        throw KMPError.fileSystem
+        set.append(contentsOf: installableFonts)
+
+        // (resource, source-destination file mapping)
+        return (resource, set)
```

Comment:
(Right-side only; Git's having trouble lining things up.)

This block takes in one `LanguageResource` - that is, one pairing between (1) either a keyboard or lexical model and (2) one language code, returning the set of files that must be installed for that pairing to be available within the KeymanEngine.  It's then installed later within the method.

The end goal, after another PR or two - `ResourceFileManager.install(resource: LanguageResource, from package: KeymanPackage)`.  Will also need the actual installation code beneath it, but the idea hopefully comes across well. 

The idea:  when a user finishes selecting from among the entries of [`KeymanPackage.installableResourceSets`](https://github.com/keymanapp/keyman/pull/3206/files?file-filters%5B%5D=.swift#diff-ad6700e6dc26f19f2833a16c9bfdf442R70), this would be used to install precisely the set of selections the user made.  Or, if the user followed a keyman.com search link or used a QR code, the link's embedded selection could be used without prompting the user.

Our legacy `parseKbdKMP` and `parseLMKMP` would auto-select the defaults from the available grouping that they currently do, then utilize that `install` method.